nsxt/resource_nsxt_policy_security_policy_rule.go

```diff
@@ -0,0 +1,246 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	securitypolicies "github.com/vmware/terraform-provider-nsxt/api/infra/domains/security_policies"
+	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+)
+
+func resourceNsxtPolicySecurityPolicyRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicySecurityPolicyRuleCreate,
+		Read:   resourceNsxtPolicySecurityPolicyRuleRead,
+		Update: resourceNsxtPolicySecurityPolicyRuleUpdate,
+		Delete: resourceNsxtPolicySecurityPolicyRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtSecurityPolicyRuleImporter,
+		},
+		Schema: getSecurityPolicyAndGatewayRuleSchema(false, false, true, true),
+	}
+}
+
+func resourceNsxtPolicySecurityPolicyRuleCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	policyPath := d.Get("policy_path").(string)
+	projectID := getProjectIDFromResourcePath(policyPath)
+	domain := getDomainFromResourcePath(policyPath)
+	policyID := getPolicyIDFromPath(policyPath)
+
+	if len(d.Get("context").([]interface{})) == 0 {
+		contexts := make([]interface{}, 1)
+		ctxMap := make(map[string]interface{})
+		ctxMap["project_id"] = projectID
+		contexts[0] = ctxMap
+		d.Set("context", contexts)
+	}
+
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID2(d, m, resourceNsxtPolicySecurityPolicyRuleExistsPartial(d.Get("policy_path").(string)))
```

Comment:
lets use `policyPath` here for readibility

Reply:
Done.